RecoTracker/TkTrackingRegions/plugins/GlobalTrackingRegionWithVerticesProducer.h

```diff
@@ -91,13 +112,56 @@ class GlobalTrackingRegionWithVerticesProducer : public TrackingRegionProducer
           if (!iV->isValid()) continue;
           if (iV->isFake() && !(theUseFakeVertices && theUseFixedError)) continue;
 	  GlobalPoint theOrigin_       = GlobalPoint(iV->x(),iV->y(),iV->z());
-	  double theOriginHalfLength_ = (theUseFixedError ? theFixedError : (iV->zError())*theSigmaZVertex); 
-	  result.push_back( std::make_unique<GlobalTrackingRegion>(thePtMin, theOrigin_, theOriginRadius, theOriginHalfLength_, thePrecise, theUseMS) );
-	  if(theMaxNVertices >= 0 && result.size() >= static_cast<unsigned>(theMaxNVertices))
-	    break;
+
+          //scaling origin radius, half length, min pt for high-occupancy HI events to keep timing reasonable
+          if(theOriginRScaling || thePtMinScaling || theHalfLengthScaling){
+            //Use the unscaled radius unless one of the two conditions below is met
+            double scaledOriginRadius = theOriginRadius;
+            double scaledHalfLength   = theFixedError;
+            double scaledPtMin        = thePtMin;
+
+            //calculate nPixels (adapted from TkSeedGenerator/src/ClusterChecker.cc)
+            double nPix = 0;
+            edm::Handle<edmNew::DetSetVector<SiPixelCluster> > pixelClusterDSV;
+            ev.getByToken(token_pc, pixelClusterDSV);
+            if (!pixelClusterDSV.failedToGet()) {
+              const edmNew::DetSetVector<SiPixelCluster> & input = *pixelClusterDSV;
+              nPix = input.dataSize();
+            }
+            else{
+              edm::LogError("GlobalTrackingRegionProducerFromVertex")<<"could not get any SiPixel cluster collections of type edm::DetSetVector<SiPixelCluster>";
+              nPix = theScalingEnd+1;//if can't find collection, default to minimum radius to be safe
```

Comment:
this seems like a symptom of a configuration error.
This should be resolved with an exception, not a LogError, which is specific for a given event.

Reply:
the easiest here is just to remove the if-else and leave just the nPix computation.